File: src/processors/move.py
"""
Move data processor.

This module processes move data from the PokéAPI.
"""

import logging
from typing import Dict, Any

from ..api_client import PokeAPIClient
from ..cache import CacheManager

logger = logging.getLogger('mcpoke-server.processors.move')

class MoveProcessor:
    """
    Processor for move data.
    
    This class processes move data from the PokéAPI.
    """
    
    def __init__(self, api_client: PokeAPIClient, cache: CacheManager):
        """
        Initialize the move processor.
        
        Args:
            api_client: PokéAPI client
            cache: Cache manager
        """
        self.api_client = api_client
        self.cache = cache
    
    def get_move(self, name_or_id: str) -> Dict[str, Any]:
        """
        Get processed information about a move.
        
        Args:
            name_or_id: Name or ID of the move
        
        Returns:
            Processed move information
        """
        # This is a skeleton implementation. We'll just pass for now.
        pass
    
    def _process_move_data(self, move_data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Process raw move data.
        
        Args:
            move_data: Raw move data from the API
        
        Returns:
            Processed move information
        """
        # This is a skeleton implementation. We'll just pass for now.
        pass
    
    def _get_move_effect(self, move_data: Dict[str, Any]) -> str:
        """
        Extract the English effect from move data.
        
        Args:
            move_data: Raw move data from the API
        
        Returns:
            English effect description
        """
        # This is a skeleton implementation. We'll just pass for now.
        pass
    
    def _get_move_short_effect(self, move_data: Dict[str, Any]) -> str:
        """
        Extract the English short effect from move data.
        
        Args:
            move_data: Raw move data from the API
        
        Returns:
            English short effect description
        """
        # This is a skeleton implementation. We'll just pass for now.
        pass

File: mcpoke-server/src/processors/move.ts
/**
 * Processing functions for move data
 */
import { pokeApiClient } from '../api/pokeapi.js';
import { cacheManager } from '../cache/cache-manager.js';
import { capitalizeFirstLetter } from '../utils/formatters.js';
import { MoveResponse } from '../api/types.js';

/**
 * Processed move data structure
 */
export interface ProcessedMove {
  id: number;
  name: string;
  type: string;
  power: number | null;
  pp: number | null;
  accuracy: number | null;
  damage_class: string;
  effect: string;
  short_effect: string;
  effect_chance: number | null;
}

/**
 * Process a move response from the API
 * @param moveData The raw move data
 * @returns The processed move data
 */
export function processMoveData(moveData: MoveResponse): ProcessedMove {
  // Get English effect entries
  const effectEntry = moveData.effect_entries.find(
    entry => entry.language.name === 'en'
  );

  // Process effect text to include effect chance
  let effect = effectEntry?.effect || 'No effect information available.';
  let shortEffect = effectEntry?.short_effect || 'No short effect information available.';
  
  // Replace $effect_chance with the actual chance
  if (moveData.effect_chance !== null) {
    effect = effect.replace(/\$effect_chance/g, moveData.effect_chance.toString());
    shortEffect = shortEffect.replace(/\$effect_chance/g, moveData.effect_chance.toString());
  }

  return {
    id: moveData.id,
    name: capitalizeFirstLetter(moveData.name),
    type: capitalizeFirstLetter(moveData.type.name),
    power: moveData.power,
    pp: moveData.pp,
    accuracy: moveData.accuracy,
    damage_class: moveData.damage_class ? capitalizeFirstLetter(moveData.damage_class.name) : 'Unknown',
    effect,
    short_effect: shortEffect,
    effect_chance: moveData.effect_chance
  };
}
